manim/mobject/opengl_three_dimensions.py

```diff
@@ -0,0 +1,85 @@
+import math
+
+from ..constants import *
+from ..mobject.types.opengl_surface import OpenGLSurface
+from ..mobject.types.opengl_surface import OpenGLSurfaceGroup
+from ..mobject.types.opengl_vectorized_mobject import OpenGLVGroup
+from ..mobject.types.opengl_vectorized_mobject import OpenGLVMobject
+from ..utils.space_ops import get_norm
+from ..utils.space_ops import z_to_vector
+
+
+class OpenGLSurfaceMesh(OpenGLVGroup):
+    def __init__(
+        self,
+        uv_surface,
+        resolution=None,
+        stroke_width=1,
+        normal_nudge=1e-2,
+        depth_test=True,
+        flat_stroke=False,
+        **kwargs
+    ):
+        if not isinstance(uv_surface, OpenGLSurface):
+            raise Exception("uv_surface must be of type OpenGLSurface")
+        self.uv_surface = uv_surface
+        self.resolution = resolution if resolution is not None else (21, 21)
+        self.normal_nudge = normal_nudge
+        super().__init__(
+            stroke_width=stroke_width,
+            depth_test=depth_test,
+            flat_stroke=flat_stroke,
+            **kwargs
+        )
+
+    def init_points(self):
+        uv_surface = self.uv_surface
+
+        full_nu, full_nv = uv_surface.resolution
+        part_nu, part_nv = self.resolution
+        u_indices = np.linspace(0, full_nu, part_nu).astype(int)
+        v_indices = np.linspace(0, full_nv, part_nv).astype(int)
+
+        points, du_points, dv_points = uv_surface.get_surface_points_and_nudged_points()
+        normals = uv_surface.get_unit_normals()
+        nudge = 1e-2
+        nudged_points = points + nudge * normals
```

Comment:
This ignores the normal_nudge variable which is set in the __init__ and sets it to a constant value

Reply:
Some may call it a bug, I call it [faithful to the original](https://github.com/3b1b/manim/blob/41a02285bde71483b8ae08ee1e05c284ef627496/manimlib/mobject/three_dimensions.py#L38). Anyway I fixed it.